server/setup/config-paths.ts

```diff
@@ -33,7 +34,17 @@ export const setupDirectories = () => {
 		try {
 			createDirectoryIfNotExist(dir);
 			if (dir === SSH_PATH) {
-				chmodSync(SSH_PATH, "600");
+				/* Changing SSH Keys permission to 600 keeping the SSH folder writable */
+				spawnSync("find", [
+					SSH_PATH,
+					"-type",
+					"f",
+					"-exec",
+					"chmod",
+					"600",
+					"{}",
+					";",
+				]);
```

Comment:
I'm running in permission issues, probably this comes with something also in the creation of keys have the incorrect permissions 

![Screenshot 2024-07-26 at 11 14 36 PM](https://github.com/user-attachments/assets/27997489-093f-4d86-b88b-696eecaed02d)
![Screenshot 2024-07-26 at 11 19 46 PM](https://github.com/user-attachments/assets/e4be782b-752d-45ee-85fc-25055489f890)

After researching a litte bit it looks like this is the correct permission which needs to have each filder/folder
| Item | Permissions (Octal) | Permissions (Symbolic) |
|------|---------------------|------------------------|
| ssh directory | 700 | drwx------ |
| Private key (id_rsa, id_ed25519) | 600 | -rw------- |
| Public key (id_rsa.pub, id_ed25519.pub) | 644 | -rw-r--r-- |

Reply:
yes but it affects only at install time, so this problem is specifically for current and new instances, what we can do is just assign the 777 permission to /etc/dokploy in a non-recursive way,

But regarding to ssh keys i think the table i shared above is the correct way, and there is no issue regarding to the command we had in the installation script since we are overriding the permissions at runtime

I make this change ca0acf044571142b9b0b9bb23f7ae0b78a454fad and now the ssh keys are working fine and running without permission issues

Reply:
The problem with ssh keys is after the install script, because permission are overwritten.
Setting 777 non recursively is a solution, and we can also get rid of [this change](https://github.com/Dokploy/dokploy/pull/269/files#diff-8109c6c242ff348593913d37c4d23848413151fcfcdf79123d7a85328c4cb7f3R131-R133).

Reply:
You are right, changing the 777 to a non-recursively could work or even assigning 755 permisions, the thing is this will work for new installations, but all the current instances will need that fix to change the permissions at runtime when they starts the server

Reply:
Yeah, I edited the answer, the default is 755.
Wouldn't be better to create a migration guide and list these commands there?

Reply:
I preffer to have compatibilty and in a major version remove this, I think the ideal solution to have compatibilty is this change ca0acf044571142b9b0b9bb23f7ae0b78a454fad, because we assign the 700 permision to ssh folder and all the ssh keys are created with the right permissions 600 and 644

Reply:
Yeah, as you prefer, it's just not needed because ssh folders on existing instances have 777, which it's still ok.
So, we can just remove the -R flag from the install scripts, right?

Reply:
that's correct all the existing instances have 777 but with this change ca0acf044571142b9b0b9bb23f7ae0b78a454fad we will change the permision in runtime because we call the `setupDirectories()` at startup server.

Yeah we can remove the -R in the installation script

Reply:
Correct, but my point is that we don't need to change the permission at runtime, because both 777 and 755 (the default from the mkdir command) are fine. Anyway, we can keep the change, no problem.

I pushed the change right now. Let me know!